### PR TITLE
dev: add read committed isolation level for transaction chapter

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What is changed, added or deleted? 
Add read committed isolation level for transaction chapter.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [x] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/transactions/transaction-isolation.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ TiDB 实现了快照隔离 (Snapshot Isolation) 级别的一致性。为与 MySQ
 
 TiDB 使用 [Percolator 事务模型](https://research.google.com/pubs/pub36726.html)，当事务启动时会获取全局读时间戳，事务提交时也会获取全局提交时间戳，并以此确定事务的执行顺序，如果想了解 TiDB 事务模型的实现可以详细阅读以下两篇文章：[TiKV 的 MVCC (Multi-Version Concurrency Control) 机制](https://pingcap.com/blog-cn/mvcc-in-tikv/)，[Percolator 和 TiDB 事务算法](https://pingcap.com/blog-cn/percolator-and-txn/)。
 
-## 可重复读
+## 可重复读隔离级别（Repeatable Read）
```
Comment (lilin90) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
## 可重复读隔离级别 (Repeatable Read)
```

File: dev/reference/transactions/transaction-isolation.md [hunk 2]
```diff
@@ -51,6 +51,16 @@ commit;                         |
 
 MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见，也就是说，即使该行在事务启动后被更新过，同样可以继续更新。这种情况在 TiDB 会导致事务回滚，导致事务最终失败，而 MySQL 是可以更新成功的。MySQL 的可重复读隔离级别并非 Snapshot 隔离级别，MySQL 可重复读隔离级别的一致性要弱于 Snapshot 隔离级别，也弱于 TiDB 的可重复读隔离级别。
 
+## 读已提交隔离级别（Read Committed）
```
Comment (lilin90) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
## 读已提交隔离级别 (Read Committed)
```

File: dev/reference/transactions/transaction-isolation.md [hunk 3]
```diff
@@ -51,6 +51,16 @@ commit;                         |
...
+
+TiDB 仅在[悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)下支持读已提交隔离级别。在乐观事务模式下设置事务隔离级别为读已提交将不会生效，事务将仍旧使用可重复读隔离级别。
+
+由于历史原因，当前主流数据库的读已提交隔离级别本质上都是 Oracle 定义的一致性读隔离级别。TiDB 为了适应这一历史原因，悲观事务中的读已提交隔离级别的实质行为也是一致性读。
+
+### 与 MySQL 读已提交隔离级别的区别
+
+MySQL 的读已提交隔离级别大部分符合一致性读特性，但其中存在某些特例，如半一致性读（[semi-consistent read](https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html)），TiDB 没有兼容这个特殊行为。
```
Comment (lilin90) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
MySQL 的读已提交隔离级别大部分符合一致性读特性，但其中存在某些特例，如半一致性读 ([semi-consistent read](https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html))，TiDB 没有兼容这个特殊行为。
```

**Comment by jackysp** [comment]
> Some minor punctuation updates. Rest LGTM.
> FYI. 如果括号里全部为英文内容，则采用英文括号；如果括号里为中文或中英混杂，则采用中文括号。

I think if it is a Chinese document, all Chinese brackets should be used, unless there is a paragraph that quotes the English original, and the brackets there should keep the same.

**Comment by lilin90** [comment]
> > Some minor punctuation updates. Rest LGTM.
> > FYI. 如果括号里全部为英文内容，则采用英文括号；如果括号里为中文或中英混杂，则采用中文括号。
> 
> I think if it is a Chinese document, all Chinese brackets should be used, unless there is a paragraph that quotes the English original, and the brackets there should keep the same.

I can understand your thoughts. The rules we adopt now are one of the commonly accepted rules for text including both Chinese and English characters.

**Comment by jackysp** [comment]
OK, @lilin90 PTAL

**Comment by jackysp** [comment]
PTAL @coocood , thx!

**Comment by coocood** [comment]
LGTM

**Pull request merged**